yjit/src/codegen.rs

```diff
@@ -516,20 +514,19 @@ pub fn jit_ensure_block_entry_exit(jit: &mut JITState, ocb: &mut OutlinedCb)
 // parameters.  When a function with optional parameters is called, the entry
 // PC for the method isn't necessarily 0, but we always generated code that
 // assumes the entry point is 0.
-fn gen_pc_guard(cb: &mut CodeBlock, iseq: IseqPtr)
+fn gen_pc_guard(cb: &mut CodeBlock, iseq: IseqPtr, insn_idx: u32)
 {
     //RUBY_ASSERT(cb != NULL);
 
     let pc_opnd = mem_opnd(64, REG_CFP, RUBY_OFFSET_CFP_PC);
+    let expected_pc = unsafe { rb_iseq_pc_at_idx(iseq, 0) };
```

Comment:
Shouldn't this be the following?
```suggestion
    let expected_pc = unsafe { rb_iseq_pc_at_idx(iseq, insn_idx) };
```